pkg/lockfile/parse-requirements-txt.go

```diff
@@ -167,6 +167,7 @@ func parseRequirementsTxt(f DepFile, requiredAlready map[string]struct{}) ([]Pac
 		}
 
 		detail := parseLine(line)
+		detail.Source = f.Path()
```

Comment:
Thanks for adding a comment. Another question -- How will `Source` be used by consumers?

Assuming this is needed by consumers, I wonder if we should just be consistent and **always** include this field?

Reply:
We probably should always include this field. it is really only optional because 

1. If Source is empty, it can be assumed to be the same value as the Lockfile (this is the case for all our lockfiles apart from requirements.txt and OSV-Scanner)
2. It'll be a lot more lines of change. 

I'm happy to go ahead and implement this for everything.

> How will Source be used by consumers?

Three examples:

1. OSV-Scanner is used in two phases. first it's ran on a system to get a full list of packages without any scanning (e.g. vm, docker container, offline/firewalled system...etc), then the osv-scanner output is loaded onto a system that can scan it later, it would be nice to preserve where each package actually comes from.  (This PR doesn't do that yet, but it provides the ability for future PR to add that capability, and enables it in the API)
2. Someone writes a custom extractor (e.g. for DEPs files) and outputs the OSV format, this allows the path to the original file to be preserved after scanning.
3. Helps find where the actual dependency is specified in lockfiles that can span multiple files (like requirements.txt)

Reply:
Got it. Makes sense now. 

Keeping it optional seems fine, and avoids a bit of bloat, but let's document this and make it explicit that it's only required in cases where it's different from the original source lockfile?

Reply:
Personally I'd actually say:
  1. if there's no immediate use-case, consider omitting it for now as it can always be added later (that's why we didn't add this when support for `-r` was added in the first place)
  2. make it required, because otherwise tools have to handle "if else" whereas if its always required then they can just use that if needed, and it shouldn't cost anything for the scanner to include it in its output

Reply:
I'd argue there is a hidden cost to including more details in the output -- it includes a little bit of cognitive load when there's more duplication in the JSON output that a human has to handle when writing code to deal with our output. 

+1 on just removing this if there are no immediate users. For the DEPs/custom extractors case, can people just re-use the existing `packageSource` field? 

i.e.
```
"packageSource": {
        "path": "/absolute/path/to/DEPS",
        "type": "custom"
      },
```

Reply:
> +1 on just removing this if there are no immediate users. For the DEPs/custom extractors case, can people just re-use the existing packageSource field?

That is where the Source field is populated from.

This `Source` field only exists in the lockfile API, and the plan is to use this in the output eventually. Happy to remove this for now and we can add this back in, in the future when we make use of it.

Reply:
> I'd argue there is a hidden cost to including more details in the output

I'd argue you're not removing that cost by making it conditional, but in fact _adding_ to it because now the human has to understand _when_ the field will actually be present; plus the conditional handling has to be forever carried around in the code and understood by every reader whereas if the field is just always valid for the most part I would expect known or ignored based on if the tool in question uses the field (i.e. when you read the code you'll either see the property being used and thus know it exists and understand it, or you won't see any reference to it in which case you don't know it exists until you do actually need it).

(related, this is why I've found `omitempty` so strange and having empty slices serialize as nil rather than an empty array - because that then requires humans to check if the output is an array or `null` instead of being able to just call their array function and have no iterations occur if its empty 🤷)

but this is getting off topic - happy to discuss separately if you'd like 🙂 

> Happy to remove this for now and we can add this back in, in the future when we make use of it.

Sounds like we're all in agreement to go with that.

Reply:
Removed source field for now :)